frontend/src/components/Button/Button.tsx

```diff
@@ -1,14 +1,30 @@
 import { ButtonHTMLAttributes } from 'react';
 
-import * as S from './style';
+import * as S from './Button.style';
+
+export type ButtonVariant = 'contained' | 'outlined' | 'text';
+type ButtonSize = 'small' | 'medium';
+type ButtonWeight = 'regular' | 'bold';
+type ButtonHoverStyle = 'base' | 'none';
 
 export interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
-  variant: 'contained' | 'outlined' | 'text';
-  size: 'small' | 'medium' | 'filled';
+  variant?: ButtonVariant;
+  size?: ButtonSize;
+  weight?: ButtonWeight;
+  hoverStyle?: ButtonHoverStyle;
+  fullWidth?: boolean;
 }
 
-const Button = ({ children, onClick, variant, size, ...rest }: Props) => (
-  <S.Button variant={variant} size={size} onClick={onClick} {...rest}>
+const Button = ({
+  children,
+  variant = 'contained',
+  size = 'medium',
+  weight = 'bold',
+  hoverStyle = 'none',
```

Comment:
```suggestion
  hoverStyle = 'base',
```

기본값을 base 로 두면 어떨까요? (none 을 사용하는 경우가 더 적을 것 같기 때문입니다!)